src/Raven.Server/Utils/Cpu/CpuUsageExtensionPoint.cs

```diff
@@ -43,124 +42,97 @@ public class CpuUsageExtensionPoint : IDisposable
-
-        private async Task ReadErrors()
-        {
-            try
-            {
-                var errors = await _process.StandardError.ReadLineAsync();
-                if (errors == null)
-                {
-                    return;
-                }
-                try
-                {
-                    if (_process.HasExited == false)
-                    {
-                        _process.Kill();
-                    }
-                }
-                catch (Exception)
-                {
-                    // When the process terminating, killed or exited
-                }
-                errors = errors + Environment.NewLine + await _process.StandardError.ReadToEndAsync();
-                NotifyWarning($"Extension point process send an error: {errors}");
-            }
-            catch (Exception e)
-            {
-                NotifyWarning("Could not read errors from cpu usage extension point process", e);
-            }
-        }
-
-        public void Dispose()
-        {
-            IsDisposed = true;
-            using (_process)
-            {
-                try
-                {
-                    if (_process.HasExited == false)
-                    {
-                        _process.Kill();
-                    }
-                }
-                catch (Exception)
-                {
-                    // When the process terminating, killed or exited
-                }
-
-                _process = null;
-            }
+                this.StartInternal(serverShutdown);
+            }, serverShutdown);
         }
 
 
-        private async Task ReadProcess()
+        private void StartInternal(CancellationToken ctk)
         {
             try
             {
-                while (IsDisposed == false)
+                var retry = 2;
+                const int maxMinutesBetweenIssues = 15;
+                var lastRestart = DateTime.UtcNow;
+                var lastReceivedLine = DateTime.UtcNow;
+                while (retry-- > 0)
                 {
-                    var nextLine = _process.StandardOutput.ReadLineAsync();
-
-                    if (await nextLine.WaitWithTimeout(_timeout) == false)
+                    using (var cts = new CancellationTokenSource())
+                    using (var linkedCts = CancellationTokenSource.CreateLinkedTokenSource(cts.Token, ctk))
                     {
-                        // here we missed an update, restart the process once
-                        HandleError($"The process didn't send information for {_timeout.TotalSeconds} seconds.");
-                        break;
-                    }
-
-                    var line = await nextLine;
-                    if (HandleInfoReceived(line) == false)
-                    {
-                        break;
+                        var lineOutHandler = new Action<object, string>((p, l) =>
+                        {
+                            Interlocked.Exchange(ref _isDataValid, 1);
```

Comment:
I can't follow what you are doing with this here, but that is suspicious and likely racy behavior